bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -102,14 +105,15 @@ public class FindReplaceOverlay extends Dialog {
 
 	private Composite replaceContainer;
 	private Composite replaceBarContainer;
-	private Text replaceBar;
+	private HistoryTextWrapper replaceBar;
 	private AccessibleToolBar replaceTools;
 	private ToolItem replaceButton;
 	private ToolItem replaceAllButton;
 
 	private Color backgroundToUse;
 	private Color normalTextForegroundColor;
 	private boolean positionAtTop = true;
+	private static final int HISTORY_SIZE = 15;
```

Comment:
Nitpicky: constants should usually be placed on top of the file (above instance fields).

Reply:
I agree, changed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/c7042b7e8f9f3aec27f4d038fa10befb832d8214